### ISSUE TITLE
Circular relationship (a module related to itself) defined as 1-M is being treated as 1-1

### ISSUE THREAD

**Issue by vjaro** [salesagility/SuiteCRM#5032]
#### Issue

I have implemented a Product - Sub Products relationship.
So i have created a 1-M relationship  AOS_Products to AOS_Products. It's working as expected the subpanel is showing and i am able to create/link sub products. But there is an issue when relating a sub product that has already been linked to other parent product.

Scenario:
3 Products. Product A, Product B, Product C

"Product A" has sub product  "Product B".
Now i want to link "Product B" to parent product "Product C"

#### Expected Behavior

"Product A" should have a sub product "Product B"
"Product C" should have a sub product "Product B"

#### Actual Behavior

"Product C" has a sub product "Product B"
"Product A" does not have a sub product "Product B" (The sub product was removed)

No suitecrm.log errors
No PHP errors

#### Steps to Reproduce

1. Create 1-M relationship Products to Products
2. Create "Product A", "Product B", "Product C", in products module
3. Link "Product B" to "Product A",  
4. Link "Product B" to "Product C", 

#### Your Environment

* SuiteCRM Version used:
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)):
* Environment name and version (e.g. MySQL, PHP 7):
* Operating System and version (e.g Ubuntu 16.04):

SuiteCRM Version 7.8.8
Google Chrome Version 63.0.3239.132 (Official Build) (64-bit)
Mysql 5.6, PHP Version 5.5.38, Apache/2.4.10
CentOS 7

**Comment by PedroErnst** [comment]
I think you will need a many to many relationship in this case.

The object in question (Product) can be taking either the left hand side or the right hand side of the relationship, therefore they must be equal.

Otherwise how would the system know when you're linking the products which side is the parent and which the child?

If you really need a hierarchy of products and sub products, then I think the only way to achieve this is to create a new module SubProducts, and create a 1-M relationship with Products.

Don't know how feasible this is for your specific scenario.

**Comment by vjaro** [comment]
@PedroErnst  i've tried the M-M approach but after a sub product was linked, when i viewed the sub product the parent product became the sub product. This might be a bug.

My only option now is to create custom module as a middle table to achieve the desired feature.

Thanks for your inputs

**Comment by PedroErnst** [comment]
Hi again @vjaro 

I've just recently had to implement something very similar, a Product and SubProduct hierarchy within the Products module.

I didn't encounter this issue you describe, I created a 1-M relationship and was able to produce a hierarchy like
```Product A
  - Product B
    - Product C```

With each of the products appearing in the subpanel of their parent product.

Admittedly the vardefs and relationships can be fairly confusing, but I don't think there is a bug here.
